Usiqee/Model/Author/Author.swift

```diff
@@ -8,13 +8,99 @@
 import FirebaseFirestoreSwift
 
 struct Author: Identifiable, Codable {
+    
+    // MARK: - Enum
+    enum SocialMedia: Int {
+        case twitter = 1
+        case facebook
+        case instagram
+        case youtube
+        case snapchat
+        case twitch
+        case tiktok
+        case linkedin
+        
+        var image: UIImage? {
+            switch self {
+            case .twitter:
+                return UIImage(named: "twitter")
+            case .facebook:
+                return UIImage(named: "facebook")
+            case .instagram:
+                return UIImage(named: "instagram")
+            case .youtube:
+                return UIImage(named: "youtube")
+            case .snapchat:
+                return UIImage(named: "snapchat")
+            case .twitch:
+                return UIImage(named: "twitch")
+            case .tiktok:
+                return UIImage(named: "tiktok")
+            case .linkedin:
+                return UIImage(named: "linkedin")
+            }
+        }
+    }
+    
+    // MARK: - Properties
     @DocumentID var id: String?
-    var name: String
-    var avatar: String
-    var desc: String
-    var twitterLink: String?
-    var fbLink: String?
-    var instagramLink: String?
-    var youtubeLink: String?
-    var webLink: String?
+    let name: String
+    let avatar: String
+    let desc: String
+    let twitterLink: String?
+    let fbLink: String?
+    let instagramLink: String?
+    let youtubeLink: String?
+    let snapchatLink: String?
+    let twitchLink: String?
+    let tiktokLink: String?
+    let linkedinLink: String?
+    let webLink: String?
+    
+    // MARK: - Helper
+    var hasExternalLinks: Bool {
+        if twitterLink == nil && fbLink == nil && instagramLink == nil && youtubeLink == nil && snapchatLink == nil && twitchLink == nil && tiktokLink == nil && linkedinLink == nil {
+            return false
+        }
+        
+        return true
```

Comment:
less optimize but better readbility
```suggestion
       !externalLinks.isEmpty
```